.github/workflows/ci.yaml

```diff
@@ -33,38 +33,51 @@ jobs:
 
   python:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
     needs: [conditional_job_checks]
     if: ${{ needs.conditional_job_checks.outputs.python_changed != 0 }}
     env:
       # TODO: remove token stored as secret in favor of using a
       # momento-local instance that can be spun up for testing
       MOMENTO_AUTH_TOKEN: ${{ secrets.ALPHA_TEST_AUTH_TOKEN }}
-
     steps:
       - uses: actions/checkout@v2
       - name: Python SDK sample
         uses: actions/setup-python@v2
         with:
-          python-version: "3.9"
-      - name: Run samples
+          python-version: ${{ matrix.python-version }}
+      - name: Run samples for python ${{ matrix.python-version }}
         id: validation
         continue-on-error: true
         run: |
           pushd python
             python -m pip install --user pipenv
-            pipenv install
+            pipenv install --python ${{ matrix.python-version }}
             pipenv run python example.py
             pipenv run python example_async.py
           popd
+      - name: Run Linters
+        id: linters
+        if: ${{ matrix.python-version == '3.7' }}
+        uses: ./.github/linters
```

Comment:
this is really cool, I like how you did this.  nitpick though, I think it might be worth putting an `actions` directory in the path to that action to make it more obvious what the contents are.  so like `./.github/actions/linters`.